Automattic/jetpack
# Writing prompts: get prompt by id when answering a specific prompt

Fixes https://github.com/Automattic/wp-calypso/issues/71938

## Proposed changes:

When answering a writing prompt, fetch by the specified prompt id, rather than using the current date.

The main benefit is that older links for past daily prompts in email or notifications will continue to work with the correct prompt.

### Other information:


## Jetpack product discussion

pe7F0s-uo-p2

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

Note: requires the wpcom patch D100649-code

- Activate the experimental editor extensions with `define( 'JETPACK_EXPERIMENTAL_BLOCKS', true );`, or the option in the Jetpack beta plugin
- Apply the diff D100649-code to your wpcom sandbox and specify your sandbox domain using the `JETPACK__SANDBOX_DOMAIN` constant.
- Specify a prompt id when creating a new post, such as `/wp-admin/post-new.php?answer_prompt=1810`
- You should see that prompt displayed as a pullquote in the editor.
- Start a new post with a non-existant prompt id, like `/wp-admin/post-new.php?answer_prompt=1`, and you should see a fallback prompt.
